### PR TITLE
V3 upgrade

### PR DESCRIPTION
**Pull request by thomheymann**
https://github.com/fastify/fastify/issues/2217

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,19 +23,20 @@
     "@types/node": "^13.1.0",
     "@typescript-eslint/parser": "^2.19.2",
     "eslint-plugin-typescript": "^0.14.0",
-    "fastify": "^2.11.0",
-    "iltorb": "2.4.3",
+    "fastify": "^3.0.0-alpha.1",
```
Comment (jsumners) [inline package.json:26]: What is the plan here, @mcollina? Are we going to release all of the core modules targeting the alpha releases and then come back and do _another_ pass to update them to the final v3?

Comment (mcollina) [reply package.json:26]: What would you prefer? I think it's the best path forward. Hopefully if dependabot works we can automate some of it.

File: package.json [hunk 2]
```diff
@@ -23,19 +23,20 @@
...
+    "iltorb": "^2.4.5",
```
Comment (mcollina) [inline package.json:27]: Can you remove this? I _think_ node 10 has native brotli.

File: index.d.ts [hunk 3]
```diff
@@ -1,27 +1,25 @@
-import { Plugin, FastifyReply, FastifyRequest } from 'fastify'
-import { Server, IncomingMessage, ServerResponse } from 'http'
+import { FastifyPlugin, FastifyReply, FastifyRequest, RawServerBase } from 'fastify'
 import { Stream } from 'stream';
+import { Input, InputObject } from 'into-stream';
```
Comment (mcollina) [inline index.d.ts:3]: I would prefer we did not expose this details to the users.

File: package.json (hunk 1 above)
Comment (jsumners) [reply package.json:26]: Sounds good. Let's learn how well dependabot helps out with this.

File: index.d.ts (hunk 3 above)
Comment (mcollina) [reply index.d.ts:3]: this is ok.

File: index.js [hunk 4]
```diff
@@ -314,6 +311,6 @@ function unzipStream (inflate, maxRecursion) {
 }
 
 module.exports = fp(compressPlugin, {
-  fastify: '>=2.11.0',
+  fastify: '>=3',
```
Comment (mcollina) [inline index.js:314]:
```suggestion
  fastify: '3.x',
```

**Pull request merged**